tanner/emulators/cmd_exec.py

```diff
@@ -0,0 +1,49 @@
+import asyncio
+import docker
+import re
+import urllib.parse
+
+from tanner.utils import patterns
+
+class CmdExecEmulator:
+	def __init__(self):
+		self.docker_client = docker.from_env(version='auto')
+
+	async def get_container(self, container_name):
+		container_if_exists = self.docker_client.containers.list(all = True,
+																 filters = dict(name = container_name)
+																 )
+		container = None
+		if container_if_exists:
+			container = container_if_exists[0]
+		return container
+	
+	async def create_attacker_env(self, session):
+		container_name = 'attacker_' + session.sess_uuid.hex
+		container = await self.get_container(container_name)
+		if not container:
+			container = self.docker_client.containers.create(image = 'ubuntu:latest',
+															 stdin_open = True, 
+															 name = container_name
+															 )
+		session.associate_env(container_name)
+		return container
+
+	async def get_cmd_exec_results(self, container, cmd):
+		container.start()
```

Comment:
I'm making new container if it doesn't exist. I think you want to ask when image don't exists.